### PR TITLE
Enables linting

### PR DESCRIPTION
**Pull request by ammarhaiderbjss**
## Description and relevant Jira numbers
- Removes prettier from the project and relies solely on tslint.
- Adds agreed rules around use of quotes, indentation and max line length
- Adds a `precommit` hook to run linting autofixes, lint and then a silent typescript compile to ensure code is valid.
- Ran tslint autofix on the existing code
- Makes some manual linting fixes

**Pro tip**: To autofix linting errors on-save, add the following flag to your user settings in VS Code [⌘,] -
`"tslint.autoFixOnSave": true`

## Pull Request checklist:

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist:

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/pages/journal/journal.ts [hunk 1]
```diff
@@ -18,11 +11,8 @@ import { AuthenticationProvider } from '../../providers/authentication/authentic
 import * as journalActions from './journal.actions';
 import { StoreModel } from '../../common/store.model';
 import {
-  getError,
-  getIsLoading,
-  getLastRefreshed,
-  getLastRefreshedTime,
-  getSlotsOnSelectedDate,
+  getError, getIsLoading, getLastRefreshed,
```
Comment (johnmorleybjss) [inline src/pages/journal/journal.ts]: Are we planning on allowing multi+multi imports? Is there a limit of imports per line?

Comment (ammarhaiderbjss) [reply src/pages/journal/journal.ts]: Yes. No, apart from the 120 line length limit

File: src/providers/slot/slot.ts [hunk 2]
```diff
@@ -77,7 +81,7 @@ export class SlotProvider {
     return Object.keys(slots).slice(0, daysAhead).reduce((acc: {[k: string]: SlotItem[]}, date) => ({
       ...acc,
       [date]: slots[date],
-    }), {});
+    }),                                                  {});
```
Comment (szabolcs-keresztes) [inline src/providers/slot/slot.ts]: Is this intentional?

Comment (ammarhaiderbjss) [reply src/providers/slot/slot.ts]: Interesting, this is happening because of the tslint rule to align arguments, looks a bit silly tbh, let me fix it

**Pull request merged**